document/document.go

```diff
@@ -86,16 +86,108 @@ func (fb FieldBuffer) GetByField(field string) (Value, error) {
 	return Value{}, ErrFieldNotFound
 }
 
+// SetFieldValue replaces a field if it already exists or creates one if not.
+func (fb *FieldBuffer) SetFieldValue(field string, reqValue Value) error {
+	_, err := fb.GetByField(field)
+	switch err {
+	case ErrFieldNotFound:
+		fb.Add(field, reqValue)
+		return nil
+	case nil:
+		_ = fb.Replace(field, reqValue)
+		return nil
+	}
+
+	return err
+}
+
+// SetValueFromValue deep replaces or creates a field
+// through the value path to get the value to set or replace.
+func (fb *FieldBuffer) SetValueFromValue(v Value, p ValuePath, newValue Value) (Value, error) {
```

Comment:
Whenever a method starts with an uppercase letter, it means that it's exported and can be used by someone else's package. It is really important to choose carefully what we want to expose.
For example here, based on the explanation, what is the difference between this method and `Set`?
As a user of this package, I only need to be able to set a value at a given path, and `Set` seems like the right choice for that, why would I need to use `SetValueFromValue`?

This method should:
- be unexported: this logic is internal to this package and shouldn't be exposed.
- be named correctly: what exactly are you trying to do with it? Something like `setValueAtPath` that would be better

Reply:
I am agree about the naming `setValueAtPath ` will be better and not exported effectively.